luarules/gadgets/game_unit_market.lua

```diff
@@ -149,19 +149,37 @@ local function offerUnitForSale(unitID, sale_price, msgFromTeamID)
     end
 end
 
+local t2cons = {"armack", "armacv", "armaca", "armacsub", "corack", "coracv", "coraca", "coracsub"}
+function contains(table, value)
+    for _, v in pairs(table) do
+        if v == value then return true end
+    end
+    return false
+end
```

Comment:
Create a table of unitDefIDs instead, keyed using the id returns true. rather than getting the unit name and going through the entire table.

Reply:
ohh it's defined in the project. I'm not the most familiar with lua, couldn't believe that such a basic function didn't exist already. thanks for the heads up sautron.

added changes #4010